spec/bitcoin.did

```diff
@@ -0,0 +1,69 @@
+type Satoshi = nat64;
+
+type OutPoint = record {
+  txid : blob;
+  vout : nat32
+};
+
+type Utxo = record {
+  outpoint: OutPoint;
+  value: Satoshi;
+  height: nat32;
+};
+
+type Pagination = record {
+  tip_hash : BlockHash;
+  offset: opt nat32;
+  limit: opt nat32;
+}
+
+type GetUtxosRequest = record {
+  address : text;
+  filter: opt variant {
+    MinConfirmations: nat32;
+    Pagination: Pagination;
+  };
+};
+
+type GetUtxosError = variant {
+  MalformedAddress;
+  // More error types to be added here.
+};
+
+type GetBalanceRequest = record {
+  address : text;
+  min_confirmations: opt nat32;
+};
+
+type GetBalanceError = variant {
+  MalformedAddress;
+  // More error types to be added here.
+};
+
+type SendTransactionError = variant {
+  MalformedTransaction;
+  // More error types to be added here.
+};
+
+service bitcoin : {
+
+  get_balance: (GetBalanceRequest) -> (variant {
+    Ok : Satoshi;
+    Err: opt GetBalanceError;
+  });
+
+  get_utxos: (GetUtxosRequest) -> (variant {
+    Ok : record {
+      utxos: vec Utxo;
+      total_count: nat32;
+      tip_hash: BlockHash;
+      tip_height: nat32;
+    };
+    Err : opt GetUtxosError;
+  });
+
+  send_transaction: (raw_transaction: blob) -> (variant {
```

Comment:
Ah, we've update this in the implementation to the following, as it makes it easier to add fields later:

```
  send_transaction: (SendTransactionRequest) -> (variant {
    Ok;
    Err : opt SendTransactionError;
  });
```

where `SendTransactionRequest` is:

```
type SendTransactionRequest = record {
  transaction: blob;
};
```